### PR TITLE
fix:  Healthcare and journal entry fixes

### PR DESCRIPTION
**Pull request by radhikagattani183**
**Healthcare - Clinical Procedure Template** - Auto fetching of item code from template name gives item not found error. #20526
**Description :** While creating Clinical Procedure Template, when we mention the template name then it auto fetches in item code and description, whether the item of that template name is present or not.

**Journal Entry** - Negative values are accepted for debit and credit amount fields. #20101
**Description :** While creating Journal Entry, when in accounts child table if we enter negative values for debit and credit amount then journal entry gets created without showing any valid error message.

**Payment Entry** - Payment entry gets created without having references. #20183
**Description :**   Payment entry should not be created without any Payment References.

**Student Admission -** Student Admission accepting end date less than start date 
**Description:** End date of admission should not be accepted less than start date.

### CODE REVIEW

File: erpnext/accounts/doctype/payment_entry/payment_entry.js [hunk 1]
```diff
@@ -923,6 +923,12 @@ frappe.ui.form.on('Payment Entry', {
 				}
 			});
 		}
+	},
+
+	validate:function(frm) {
+		if (frm.doc.references.length === 0) {
```
Comment (nabinhait) [inline erpnext/accounts/doctype/payment_entry/payment_entry.js:929]: Why so? Reference is not mandatory in payment entry for advance payment

File: erpnext/accounts/doctype/journal_entry/journal_entry.js [hunk 2]
```diff
@@ -451,10 +451,20 @@ frappe.ui.form.on("Journal Entry Account", {
 	},
 
 	debit_in_account_currency: function(frm, cdt, cdn) {
+		var row = locals[cdt][cdn];
+		if(row.debit_in_account_currency < 0 || row.debit_in_account_currency === 0){
+			frappe.model.set_value(cdt, cdn, "debit_in_account_currency", '');
+			frappe.throw(__("Debit Amount cannot be zero or negative."));
```
Comment (nabinhait) [inline erpnext/accounts/doctype/journal_entry/journal_entry.js:457]: Debit / credit amount can be zero

File: erpnext/accounts/doctype/journal_entry/journal_entry.js [hunk 3]
```diff
@@ -451,10 +451,20 @@ frappe.ui.form.on("Journal Entry Account", {
 	},
 
 	debit_in_account_currency: function(frm, cdt, cdn) {
+		var row = locals[cdt][cdn];
+		if(row.debit_in_account_currency < 0 || row.debit_in_account_currency === 0){
+			frappe.model.set_value(cdt, cdn, "debit_in_account_currency", '');
```
Comment (nabinhait) [inline erpnext/accounts/doctype/journal_entry/journal_entry.js:456]: why you are setting it as blank?

File: erpnext/education/doctype/student_admission/student_admission.js [hunk 4]
```diff
@@ -11,5 +11,11 @@ frappe.ui.form.on('Student Admission', {
 
 	academic_year: function(frm) {
 		frm.trigger("program");
+	},
+	admission_end_date: function(frm) {
+		if(frm.doc.admission_end_date && frm.doc.admission_end_date <= frm.doc.admission_start_date){
+			frm.set_value("admission_end_date", "");
+			frappe.throw(__("Please select the valid <b>Admission End Date</b>"));
```
Comment (nabinhait) [inline erpnext/education/doctype/student_admission/student_admission.js]:
```suggestion
			frappe.throw(__("Admission End Date should be after Admission Start Date"));
```

**Review by nabinhait — CHANGES_REQUESTED** [review]

File: erpnext/accounts/doctype/journal_entry/journal_entry.js (hunk 3 above)
Comment (radhikagattani183) [reply erpnext/accounts/doctype/journal_entry/journal_entry.js:456]: On considering, Journal Entry can't be created with 0 credit/debit amount. Hence set it to blank.

File: erpnext/accounts/doctype/journal_entry/journal_entry.js (hunk 2 above)
Comment (radhikagattani183) [reply erpnext/accounts/doctype/journal_entry/journal_entry.js:457]: Then what is the use of creating JV with zero amount.?

File: erpnext/accounts/doctype/payment_entry/payment_entry.js (hunk 1 above)
Comment (radhikagattani183) [reply erpnext/accounts/doctype/payment_entry/payment_entry.js:929]: Then we have to handle the scenario where PO is created with on_hold flag and PE is created against that.

File: erpnext/accounts/doctype/journal_entry/journal_entry.js (hunk 2 above)
Comment (nabinhait) [reply erpnext/accounts/doctype/journal_entry/journal_entry.js:457]: In a specific row, either debit or credit will have value, other will be zero (or blank).

File: erpnext/accounts/doctype/payment_entry/payment_entry.js (hunk 1 above)
Comment (nabinhait) [reply erpnext/accounts/doctype/payment_entry/payment_entry.js:929]: Raise a separate PR for that scenario

**Review by nabinhait — COMMENTED** [review]
@radhikagattani123 I think, accounts related changes does not make any sense. Can you make a separate PR only for healthcare chages? Closing this.

**Pull request closed without merging**